lib/events/athena/consumer.go

```diff
@@ -302,7 +298,15 @@ func (c *consumer) processBatchOfEvents(ctx context.Context) (reachedMaxSize boo
 	go func() {
 		msgsCollector.fromSQS(readSQSCtx)
 	}()
-	toDelete, err := c.writeToS3(ctx, msgsCollector.getEventsChan(), c.perDateFileParquetWriter)
+
+	fanOutEventsChan := make(chan eventAndAckID, c.batchMaxItems)
+	// metricsChan will be processed without any blocking, no need for big buffer.
+	metricsChan := make(chan eventAndAckID, 100)
```

Comment:
Do we really need do it async ? 
It just a few inc/dec/ if calls that should be insignificant in terms of performance.

Reply:
I initially started with few inc/dec/ if calls but was not happy with readability. I give it another try in https://github.com/gravitational/teleport/pull/26331/commits/29998bbc2eb23c38a06621fb72e7c666ba8e3320 and I think now it's more readable. I have dropped that sync solution.